include/zephyr/net/net_filter.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Public API for network filter
+ *
+ * The simple network filter allows to network-related operations
+ * which can be implemented in the form of hooks - custom callbacks.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_FILTER_H_
+#define ZEPHYR_INCLUDE_NET_FILTER_H_
+
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/net_pkt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Network Filter API
+ * @defgroup net_filter Network Filter API
+ * @ingroup networking
+ * @{
+ */
+
+/** @brief hook callback type. */
+typedef enum net_verdict (nf_hook_fn_t)(struct net_pkt *pkt);
+
+/** @brief Hook IDs for IP. */
+typedef enum {
+	NF_IP_PRE_ROUTING = 0,
+	NF_IP_LOCAL_IN,
+	NF_IP_FORWARD,
+	NF_IP_LOCAL_OUT,
+	NF_IP_POST_ROUTING,
+	NF_IP_NUMHOOKS
+} nf_ip_hooks;
+
+/** @brief Hook configuration structure with metadata.
+ *
+ * Several callback functions can be registered with the same hook.
+ * The priority value determines the order of callbacks function and
+ * their calling order. The callback function for which priority has
+ * been set to lower value (e.g. -100) will be called before callback
+ * function with higher priority value (e.g. 0).
+ */
+struct nf_hook_cfg {
+	nf_hook_fn_t *hook_fn;  /**< callback function */
+	unsigned int hooknum;   /**< hook ID number */
```

Comment:
There's a dedicated enum for the value defined just above. Also, I'd prefer `hook type` instead of `hook ID number`.

Reply:
OK, I changed description to `hook type category` however I want to avoid assign specific type for this value. When this module evaluate in future then can be different hooks categories for e.g. [brige](https://en.wikipedia.org/wiki/Netfilter#/media/File:Netfilter-packet-flow.svg): `brouting`